### ISSUE TITLE
Display RTMP don't fit screen

### ISSUE THREAD

**Issue by linhlq58** [pedroSG94/rtmp-rtsp-stream-client-java#819]
I'm working on an app which is live-streaming on Facebook using Display RTMP. But I noticed that the output video don't fit the screen, there are black spaces on 2 sides. I think this issue happens when screen size of device does not match with resolution (720 x1080).

Is there any way to just capture a part of screen and stream to Facebook to avoid this issue?

![photo_2021-04-02_18-06-40](https://user-images.githubusercontent.com/10673572/113410884-a24e5f00-93de-11eb-9642-ee253952661f.jpg)

**Comment by pedroSG94** [comment]
Hello,

Display mode support any resolution. If you select a resolution different of device screen MediaProjectionManager scale it for you.

**Comment by linhlq58** [comment]
@pedroSG94 Yes, but the output video will not be at fullscreen. I want a fullscreen video without any black spaces. Are there any options to trim screen before streaming?

**Comment by pedroSG94** [comment]
This is not a bug, I tried record screen with native oneplus app and the result is the same (black spaces).
I suspect that mediaprojection api is doing it (I mean it is already full screen).
We can apply a filter to crop it. Knowning it the question is:
- How to crop that spaces if I don't know the size? Do you have any sugestion?

I will re check it another time to comfirm all

**Comment by linhlq58** [comment]
Okay 👯  Please let me know when you have a solution. I really need your help.

**Comment by pedroSG94** [comment]
Ok, I found a way to fit screen but this produce image distortion to adapt it (it is normal you can't fit without distortion using different resolutions).
Go to this line and set width / height to screen device resolution:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/blob/master/rtplibrary/src/main/java/com/pedro/rtplibrary/base/DisplayBase.java#L392
Also in this line:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/blob/master/encoder/src/main/java/com/pedro/encoder/input/gl/render/CameraRender.java#L68

**Comment by linhlq58** [comment]
@pedroSG94 Thank you for your solution. I tried and it worked. But it's a bad UX in a livestream app if images are distorted. Is there anyway to fit screen without distortion  by cutting the top edge or bottom edge of the screen while streaming?

**Comment by pedroSG94** [comment]
Yes, that is possible but, for now, I will mark it as feature because it will take time to develop properly.

**Comment by hoangcongtuan** [comment]
Hi @linhlq58, do you have any solution for this?
